alerts/cluster-kube-controller-manager-operator/GarbageCollectorSyncFailed.md

```diff
@@ -0,0 +1,71 @@
+# GarbageCollectorSyncFailed
+
+## Meaning
+
+[This alert][GarbageCollectorSyncFailed] is fired when some resources couldn't
+be synced for a garbage collector informer cache. This usually happens due to
+the failing conversion webhook of an installed `CustomResourceDefinition` (CRD)
+or because of failing API Discovery / problems with network.
+
+## Impact
+
+Garbage collection and orphaning of objects stops working. Normal deletion
+works fine, but deletion of objects via owner references does not.
+Deletion of an object with `--cascade=orphan` option also does not work.
+
+This can lead to resource exhaustion or/and etcd pollution
+when objects are not garbage collected.
+
+For example:
+- replica sets can leave running pods behind
+- cron jobs can leave completed pods behind
+
+In extreme cases this can lead to cluster failure.
+
+
+## Diagnosis
+
+Analyze logs of `kube-controller-manager` pods in
```

Comment:
Probably worth noting down that other than this alert this can be noticed through pods not being removed (stuck in terminating phase) in response to removing deployment, for example.

Reply:
I have something similar in the Impact section. Not sure if I should repeat it here as well? Or move that part to diagnosis?

Reply:
Maybe clarify that removing replica set will leave pods behind, because your current wording:
```
- replica sets can leave running pods behind
- cron jobs can leave completed pods behind
```
suggests the controllers leave the pods behind, which is not true, it's the removal of the controllers should normally also trigger pod removal, but it doesn't happen.

Reply:
fixed